client/components/data/document-head/index.jsx

```diff
@@ -32,39 +31,39 @@ class DocumentHead extends Component {
 		if ( this.props.meta !== undefined ) {
 			this.props.setMeta( this.props.meta );
 		}
-	}
 
-	componentDidMount() {
 		this.setFormattedTitle( this.props.formattedTitle );
 	}
 
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
+	componentDidUpdate( prevProps ) {
 		// The `title` prop is commonly receiving its value as a result from a `translate` call
 		// and in some cases it returns a React component instead of string.
 		// A shallow comparison of two React components may result in unnecessary title updates.
 		// To avoid that, we compare the string representation of the passed `title` prop value.
 		if (
-			nextProps.title !== undefined &&
-			this.props.title?.toString?.() !== nextProps.title?.toString?.()
+			this.props.title !== undefined &&
+			prevProps.title?.toString?.() !== this.props.title?.toString?.()
 		) {
-			this.props.setTitle( nextProps.title );
+			prevProps.setTitle( this.props.title );
```

Comment:
Is there a reason we need to use previous props in those cases or could we use `this.props.setTitle` instead? I think that'd make the code more reasonable if there's no requirement we have to use the previous prop for these cases. Question applies to other functions used in this lifecycle hook as well.